drivers/video/video_esp32_dvp.c

```diff
@@ -324,6 +358,23 @@ static int video_esp32_get_ctrl(const struct device *dev, unsigned int cid, void
 	return video_get_ctrl(cfg->source_dev, cid, value);
 }
 
+#ifdef CONFIG_POLL
+int video_esp32_set_signal(const struct device *dev, enum video_endpoint_id ep,
+			   struct k_poll_signal *signal)
+{
+	struct video_esp32_data *data = dev->data;
+
+	if (ep == VIDEO_EP_OUT) {
+		data->signal_out = signal;
+		return 0;
+	}
+
+	LOG_ERR("Invalid endpoint id");
+
+	return -EINVAL;
```

Comment:
https://github.com/zephyrproject-rtos/zephyr/pull/77770#discussion_r1738728176 suggests that the preferred way is to reject the invalid cases rather than accepting the valid ones.

Also, `VIDEO_EP_ALL` sounds like a valid request that should affect all endpoints.

```suggestion
	if (ep != VIDEO_EP_OUT && ep != VIDEO_EP_ALL) {
		LOG_ERR("Invalid endpoint id");
		return -EINVAL;
	}

	data->signal_out = signal;
	return 0;
```

This is a bug in the current driver. Feel free to only keep `VIDEO_EP_OUT` if you prefer, this would get all fixed at once on a separate PR.
- https://github.com/zephyrproject-rtos/zephyr/issues/78712